sqlmesh/core/engine_adapter/base.py

```diff
@@ -604,9 +604,14 @@ def _create_table_from_source_queries(
         # types, and for evaluation methods like `LogicalReplaceQueryMixin.replace_query()`
         # calls and SCD Type 2 model calls.
         schema = None
+        columns_to_types_all_known = columns_to_types and all(
```

Comment:
can you make the all() check into a helper method?

delete the extra columns_to_types check on line 613

Reply:
can you add a test for this?

Reply:
Done. The check on line 613 was for mypy - i added a type ignore where it complained instead.